pkg/cache/context.go

```diff
@@ -700,17 +700,24 @@ func (ctx *Context) RemoveTask(appID, taskID string) error {
 	return fmt.Errorf("application %s is not found in the context", appID)
 }
 
-func (ctx *Context) getTask(appID string, taskID string) (*Task, error) {
+func (ctx *Context) getTask(appID string, taskID string) *Task {
 	ctx.lock.RLock()
 	defer ctx.lock.RUnlock()
-	if app, ok := ctx.applications[appID]; ok {
+	if app := ctx.GetApplication(appID); app != nil {
```

Comment:
Looking at what we log in case of an issue I think we have a flow issue. If the task conversion fails we log 3 things: conversion failed, task not found and app not found. Really confusing. Go coding guidelines also state to prefer to return early and do not indent the correct case, indent the error handling
That would get a better user experience would be to just log the most specific message.
```
app := ctx.GetApplication(appID)
if app == nil {
    log.Logger().....
    return nil
}
managedTask, err := app.GetTask(taskID)
if err != nil {
    log.Logger().....
    return nil
}
task, valid := managedTask.(*Task)
if !valid {
    log.Logger().....
    return nil
}
return task
```

Reply:
OK, that is really more clean, thank you